src/rgw/rgw_op.cc

```diff
@@ -691,24 +671,21 @@ int rgw_build_bucket_policies(const DoutPrefixProvider *dpp, rgw::sal::Driver* d
 int rgw_build_object_policies(const DoutPrefixProvider *dpp, rgw::sal::Driver* driver,
 			      req_state *s, bool prefetch_data, optional_yield y)
 {
-  int ret = 0;
-
-  if (!rgw::sal::Object::empty(s->object.get())) {
-    if (!s->bucket_exists) {
-      return -ERR_NO_SUCH_BUCKET;
-    }
-    s->object_acl = std::make_unique<RGWAccessControlPolicy>();
+  if (rgw::sal::Object::empty(s->object)) {
```

Comment:
this is different logic than the original code. is this the intent?
previously, when there was no object and no bucket, we would have returned `ERR_NO_SUCH_BUCKET`, now we return `0`

Reply:
> previously, when there was no object and no bucket, we would have returned ERR_NO_SUCH_BUCKET

when there was no object, `if (!rgw::sal::Object::empty(s->object.get())) {` evaluated to `if (!true)`, so the rest of the function (including the check for `!bucket_exists`->`ERR_NO_SUCH_BUCKET`) was skipped

so i *think* the new logic is equivalent to the old, i just inverted the condition for an early-return